codec/amino_codec.go

```diff
@@ -1,18 +1,24 @@
 package codec
 
 import (
+	"fmt"
+
 	"github.com/cosmos/gogoproto/proto"
+	protov2 "google.golang.org/protobuf/proto"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
 )
 
-// AminoCodec defines a codec that utilizes Codec for both binary and JSON
-// encoding.
+// Deprecated: AminoCodec defines a codec that utilizes Codec for both binary and JSON
+// encoding. Any usage of amino should be done using the LegacyAmino type directly.
+// Usage of amino with the Codec type is not well-supported and may be removed in the future.
 type AminoCodec struct {
 	*LegacyAmino
 }
 
 var _ Codec = &AminoCodec{}
 
-// NewAminoCodec returns a reference to a new AminoCodec
+// Deprecated: NewAminoCodec returns a reference to a new AminoCodec
```

Comment:
can we mention what this is deprecated in favour of?

Reply:
updated docs

Reply:
No LegacyAmino, which doesn't attempt to implement the Codec interface